lib/wallet/src/Cardano/Wallet/Write/Tx/Balance.hs

```diff
@@ -1072,35 +1071,12 @@ assignChangeAddresses (ChangeAddressGen genChange _) sel = runState $ do
 posAndNegFromCardanoValue
     :: Cardano.Value
     -> (TokenBundle.TokenBundle, TokenBundle.TokenBundle)
-posAndNegFromCardanoValue = foldMap go . Cardano.valueToList
-  where
-    go :: (Cardano.AssetId, Cardano.Quantity)
-       -> (TokenBundle.TokenBundle, TokenBundle.TokenBundle)
-    go (Cardano.AdaAssetId, q) = partition q $
-        TokenBundle.fromCoin . Coin.fromNatural
-    go (Cardano.AssetId policy name, q) = partition q $ \n ->
-        TokenBundle.fromFlatList (W.Coin 0)
-            [ ( TokenBundle.AssetId (mkPolicyId policy) (mkTokenName name)
-              , TokenQuantity n
-              )
-            ]
-
-    -- | Convert a 'Cardano.Quantity' to a 'TokenBundle' using the supplied
-    -- function. The result is stored in 'fst' for positive quantities, and
-    -- 'snd' for negative quantities.
-    partition
-        :: Cardano.Quantity
-        -> (Natural -> TokenBundle.TokenBundle)
-        -> (TokenBundle.TokenBundle, TokenBundle.TokenBundle)
-    partition (Cardano.Quantity i) f
-        | Just n <- maybeIntegerToNatural      i  = (f n, mempty)
-        | Just n <- maybeIntegerToNatural (abs i) = (mempty, f n)
-        | otherwise = (mempty, mempty)
-
-    maybeIntegerToNatural = intCastMaybe @Integer @Natural
-
-    mkPolicyId = UnsafeTokenPolicyId . Hash . Cardano.serialiseToRawBytes
-    mkTokenName = UnsafeTokenName . Cardano.serialiseToRawBytes
+posAndNegFromCardanoValue
+    = bimap
+        (fromCardanoValue . Cardano.valueFromList)
+        (fromCardanoValue . Cardano.valueFromList . L.map (second negate))
```

Comment:
Noticing that `fromCardanoValue . Cardano.valueFromList` is repeated twice

```haskell
    bimap
        (fromCardanoValue . Cardano.valueFromList)
        (fromCardanoValue . Cardano.valueFromList . L.map (second negate))
```
it could be moved it up

```haskell
    fromCardanoValue . Cardano.valueFromList
        . bimap identity (L.map (second negate))
```
and then 
```haskell
 fromCardanoValue . Cardano.valueFromList . fmap (fmap (second negate))
```